### PR TITLE
Service catalog minimal implementation

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Minimal implementation of the new service catalog.

image

---

Below are the tasks I plan to add to the backlog.
This list is based on the specifications discussed here: https://outline.teclib.com/doc/catalogue-de-service-interface-self-service-7OGd1p5dVx
Changes to ticket validation will be discussed in separate specifications and are not included in the list.

Tasks:
* Filter forms according to access policies
* Limit the number of forms that can be displayed (pagination)
* Add search bar
* Add form icons
* Add form descriptions (currently we display form header instead, which is not meant to be used for this)
* Add forms accent colors
* Add categories
* Add "pinned" forms 
* Replace "Tickets" menu entry by multiple entries ("Ongoing tickets", "Closed tickets", ...).
* Add unread messages counters to menus
* Final service catalog UI/UX review
* Create a default form on install / update to 11 that ask the same questions as the old helpdesk html form.
* Force creation of at least one ticket per form
* Add "auto close" option to this mandatory ticket (to get a similar behavior that form without targets in formcreator)

Let me know if you think something is missing from this list.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

**Comment by AdrienClairembault** [comment]
Draft as some ongoing PR must be merged first.

File: front/helpdesk.public.php
Comment (AdrienClairembault) [inline front/helpdesk.public.php:1]: Diff is much easier to review with white space changes disabled.

File: templates/layout/page_skeleton.html.twig [hunk 1]
```diff
@@ -30,7 +30,10 @@
  # ---------------------------------------------------------------------
  #}
 
-{% do call(['Html', 'header'], {
+{% set interface = call("Session::getCurrentInterface") %}
```
Comment (cedric-anne) [inline templates/layout/page_skeleton.html.twig:33]: 1. There is a Twig function for it.
2. We should avoid to use the `call` function in new features. It has been made to ease the transition to Twig but its usages have to be replaced by some cleaner code later.
```suggestion
{% set interface = get_current_interface() %}
```

File: templates/pages/self-service/service_catalog.html.twig [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                        class="card-body"
+                        aria-labelledby="form-{{ form.fields.id }}"
+                    >
+                        <h2 id="form-{{ form.fields.id }}" class="card-title">
+                            {{ form.fields.name }}
+                        </h2>
+                        {# TODO: showing description instead of header (description does not yet
+                           exist and header is meant to be displated above the form in its own
+                           page). We use header for now to "fill" the space. #}
+                        <div class="text-secondary">{{ form.fields.header|raw }}</div>
```
Comment (cedric-anne) [inline templates/pages/self-service/service_catalog.html.twig:60]: Rich text provided by end user must be rendered using the `|safe_html` filter that will clean any unsafe code.
```suggestion
                        <div class="text-secondary">{{ form.fields.header|safe_html }}</div>
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**